### DISCUSSION TITLE
conversion double to 2 decimal float

### DISCUSSION THREAD

**Discussion by ilkot** [nlohmann/json#2895] [Q&A] [answered]
Hello all,
I can convert double to 2 digit decimal with below function;

```
float round2float(double var, int decimal)
{
    int scale = std::pow(10, decimal);

    float value = (int)(var * scale + .5);
    return (float)value / scale;
}
```

but when I store it in json, it creates double with long decimals.
I looked at the number handling documentation it also refers this issue, but do you know how to overcome?

an example;
```

double var1 = 1.149999976158142;
float var2 = round2float(var1,2);
std::cout<<var2<<std::endl;
```
prints
`1.15`

```

nlohmann::json varJson = {{"var",var2}};
std::cout<<varJson<<std::endl;

```
prints
`[{"var":1.149999976158142}]`

So, is there way that I can obtain [{"var":1.15}] ?

Thanks in advance.

**Comment by nlohmann** [comment, marked as answer]
No, this is currently not possible.

**Reply by ilkot** [reply]
Thanks for quick answer!
